src/com/goide/inspections/GoUnusedImportDeclaration.java

```diff
@@ -62,6 +69,19 @@ protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsH
     for (GoImportSpec duplicatedImportSpec : GoImportOptimizer.findDuplicatedEntries(importMap)) {
       problemsHolder.registerProblem(duplicatedImportSpec, "Redeclared import", ProblemHighlightType.GENERIC_ERROR, OPTIMIZE_QUICK_FIX);
     }
+
+    // Find imports that are referred to by the same alias or last pathname component.
+    MultiMap<String, GoImportSpec> nameMap = getImportNameMap(file);
+    for (Map.Entry<String, Collection<GoImportSpec>> specs : nameMap.entrySet()) {
+      // These iterators are guaranteed to be non-empty.
+      Iterator<GoImportSpec> imports = specs.getValue().iterator();
+      // Add warnings to the second and subsequent imports.
+      imports.next();
+      while (imports.hasNext()) {
+        problemsHolder.registerProblem(imports.next(), "Redeclared import", ProblemHighlightType.GENERIC_ERROR, OPTIMIZE_QUICK_FIX);
```

Comment:
OptimizeImports quick fix does nothing with partially matched imports so this problem shouldn't declare it and possible fix.